consensus/consensus_validator.go

```diff
@@ -174,106 +168,41 @@ func (consensus *Consensus) processChallengeMessage(message consensus_proto.Mess
-		return
-	}
-
-	// For now, simply return the private key of this node.
-	receivedChallenge := crypto.Ed25519Curve.Scalar()
-	err = receivedChallenge.UnmarshalBinary(challenge)
-	if err != nil {
-		log.Error("Failed to deserialize challenge", "err", err)
-		return
-	}
-
-	if !reconstructedChallenge.Equal(receivedChallenge) {
-		log.Error("The challenge doesn't match the commitments and keys")
-		return
-	}
-
-	response, err := crypto.Response(crypto.Ed25519Curve, consensus.priKey, consensus.secret[consensusID], receivedChallenge)
-	if err != nil {
-		log.Warn("validator failed to generate response", "err", err, "priKey", consensus.priKey, "nodeID", consensus.nodeID, "secret", consensus.secret[consensusID])
-		return
-	}
+	_ = multiSig
+	_ = bitmap
+	// TODO: verify multi signature
 
-	msgTypeToSend := consensus_proto.MessageType_RESPONSE
-	if targetState == FinalResponseDone {
-		msgTypeToSend = consensus_proto.MessageType_FINAL_RESPONSE
-	}
-	msgToSend := consensus.constructResponseMessage(msgTypeToSend, response)
+	// Construct and send the commit message
+	msgToSend := consensus.constructCommitMessage()
 
 	consensus.SendMessage(consensus.leader, msgToSend)
-	// utils.GetLogInstance().Warn("Sending Response to leader", "state", targetState)
-	// Set state to target state (ResponseDone, FinalResponseDone)
-	consensus.state = targetState
-
-	if consensus.state == FinalResponseDone {
-		// TODO: the block catch up logic is a temporary workaround for full failure node catchup. Implement the full node catchup logic
-		// The logic is to roll up to the latest blocks one by one to try catching up with the leader.
-		for {
-			val, ok := consensus.blocksReceived[consensus.consensusID]
-			if ok {
-				delete(consensus.blocksReceived, consensus.consensusID)
-
-				consensus.blockHash = [32]byte{}
-				delete(consensus.secret, consensusID)
-				consensus.consensusID = consensusID + 1 // roll up one by one, until the next block is not received yet.
-
-				var blockObj types.Block
-				err := rlp.DecodeBytes(val.block, &blockObj)
-				if err != nil {
-					utils.GetLogInstance().Warn("Unparseable block header data", "error", err)
-					return
-				}
-				if err != nil {
-					utils.GetLogInstance().Debug("failed to construct the new block after consensus")
-				}
-				// check block data (transactions
-				if !consensus.BlockVerifier(&blockObj) {
-					utils.GetLogInstance().Debug("[WARNING] Block content is not verified successfully", "consensusID", consensus.consensusID)
-					return
-				}
-				utils.GetLogInstance().Info("Finished Response. Adding block to chain", "numTx", len(blockObj.Transactions()))
-				consensus.OnConsensusDone(&blockObj)
-
-				select {
-				case consensus.VerifiedNewBlock <- &blockObj:
-				default:
-					utils.GetLogInstance().Info("[SYNC] consensus verified block send to chan failed", "blockHash", blockObj.Hash())
-					continue
-				}
-
-			} else {
-				break
-			}
 
-		}
-	}
+	consensus.state = CommitDone
```

Comment:
Why processPrepareMessage will set the state CommitDone? Not PreparedDone?

Reply:
I will correct this in a following PR.